### PR TITLE
Add a test case with dropped categorical features

### PR DESCRIPTION
**Pull request by romanlutz**
So far we had no test case with dropped features. Since there was a bug with dropped categorical features in v0.7.0 it's prudent to add a test case for this. This way, we can catch future regressions.

### CODE REVIEW

File: src/responsibleai/rai_analyse/create_score_card.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def validate_and_correct_config(config, insight_data):
             f for f in fc["sensitive_features"] if f not in i_data.categorical_features
         ]
         _logger.warning(
-            "Non categorical features in fairness dropped: {}".format(dropped_features)
+            "Non categorical features dropped for fairness assessment: {}".format(dropped_features)
```
Comment (romanlutz) [inline src/responsibleai/rai_analyse/create_score_card.py:150]: This is unrelated but clearly an improvement on the logged text, so I added it in this PR. If you strongly feel like this warrants its own PR please comment.

File: .gitignore [hunk 2]
```diff
@@ -367,6 +367,7 @@ MigrationBackup/
 **/*.processed.yaml
 **/*.output.ipynb
 **/*.processed.ipynb
+**/*.yaml.updated
```
Comment (hawestra) [inline .gitignore:370]: why is this added ?

**Review by imatiach-msft — APPROVED** [review]

File: test/rai/pipeline_adult_analyse.yaml [hunk 3]
```diff
@@ -25,6 +25,7 @@ jobs:
     inputs:
       training_data: ${{parent.inputs.my_training_data}}
       target_column_name: ${{parent.inputs.target_column_name}}
+      dropped_features: 'Race'
```
Comment (tongyu-microsoft) [inline test/rai/pipeline_adult_analyse.yaml:28]: should this be a list? '["Race"]'?

**Review by tongyu-microsoft — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (romanlutz) [reply .gitignore:370]: Because these files get generated when you run tests locally and they shouldn't be checked in 🙂

File: test/rai/pipeline_adult_analyse.yaml (hunk 3 above)
Comment (romanlutz) [reply test/rai/pipeline_adult_analyse.yaml:28]: It depends entirely on what we do inside the component. This is just a test component, not one we actually want people to use, so I don't know if it matters either way. I went with the human-readable comma-separated string for simplicity.
